py/server/deephaven/table.py

```diff
@@ -47,12 +48,17 @@
 
 # Dynamic Query Scope
 _JExecutionContext = jpy.get_type("io.deephaven.engine.context.ExecutionContext")
-_JQueryScope = jpy.get_type("io.deephaven.engine.context.QueryScope")
-_JUnsynchronizedScriptSessionQueryScope = jpy.get_type(
-    "io.deephaven.engine.util.AbstractScriptSession$UnsynchronizedScriptSessionQueryScope")
+_JScriptSessionQueryScope = jpy.get_type("io.deephaven.engine.util.AbstractScriptSession$ScriptSessionQueryScope")
 _JPythonScriptSession = jpy.get_type("io.deephaven.integrations.python.PythonDeephavenSession")
-_j_script_session = jpy.cast(_JExecutionContext.getContext().getQueryScope(), _JUnsynchronizedScriptSessionQueryScope).scriptSession()
-_j_py_script_session = jpy.cast(_j_script_session, _JPythonScriptSession)
+
+
+def _j_py_script_session() -> _JPythonScriptSession:
+    j_execution_context = _JExecutionContext.getContext()
+    j_query_scope = j_execution_context.getQueryScope()
+    if not j_query_scope:
+        raise DHError("ExecutionContext does not have QueryScope")
```

Comment:
check for PoisonedQueryScope?

Reply:
The following cast will throw an appropriate exception, essentially saying that a poisoned query scope can't be cast into the type we are wanting to.

Reply:
That said; this may have been me fumbling around trying to figure out why my DHError wasn't working. Looking at the code, I don't think any of the values in ExecutionContext can/should be null. I'll add a not null check to ExecutionContext, and I think we can get rid of this extraneous check.